### PR TITLE
Fix the behaviour of setResultStateHandler when shareStream parameter is true

### PR DESCRIPTION
**Pull request by PrimeSwiftArt**

### CODE REVIEW

File: lib/bloc/rx_bloc_base.dart [hunk 1]
```diff
@@ -85,7 +85,7 @@ abstract class RxBlocBase {
   @protected
   Stream<Result<T>> setResultStateHandler<T>(Stream<Result<T>> resultStream,
       {bool shareStream = true}) {
-    if (shareStream && !resultStream.isBroadcast) {
+    if (shareStream && resultStream is! Subject<Result<T>>) {
```
Comment (StanevPrime) [inline lib/bloc/rx_bloc_base.dart:88]: Could you please elaborate around the second condition?

File: test/bloc/rx_bloc_base_test.dart [hunk 2]
```diff
@@ -50,5 +50,61 @@ void main() {
...
+          .whereSuccess();
+      expect(bloc.loadingState, emitsInOrder([]));
+      expect(data, emitsInOrder([]));
+      expect(listenCount, 3);
+    });
+
+    test('Test setResultStateHandler with shareStream disabled om broadcast stream', () async {
+      int listenCount = 0;
+      final bloc = BlocImpl();
+      final stream = Future.value(3.14).asResultStream()
```
Comment (StanevPrime) [inline test/bloc/rx_bloc_base_test.dart:93]: The stream does not look to be a broad cast

File: test/bloc/rx_bloc_base_test.dart [hunk 3]
```diff
@@ -50,5 +50,61 @@ void main() {
...
+          .doOnListen(() { listenCount++; });
+      final data = bloc.setResultStateHandler(stream, shareStream: false)
+          .whereSuccess();
+      expect(bloc.loadingState, emitsInOrder([]));
+      expect(data, emitsInOrder([]));
+      expect(listenCount, 3);
+    });
+
+    test('Test setResultStateHandler with shareStream disabled om broadcast stream', () async {
```
Comment (StanevPrime) [inline test/bloc/rx_bloc_base_test.dart:90]: There is a typo in the description.

**Comment by PrimeSwiftArt** [comment]
@StanevPrime I think we have to set the package version to 0.3.0, because there is new functionality.

**Comment by StanevPrime** [comment]
Agree

**Pull request merged**